ci/mkpipeline.py

```diff
@@ -240,6 +243,32 @@ def visit(step: dict[str, Any]) -> None:
                 visit(inner_step)
 
 
+def trim_test_selection(pipeline: Any, steps_to_run: set[str]) -> None:
+    def visit(step: dict[str, Any]) -> None:
+        if (
+            step.get("id") not in steps_to_run
+            and "prompt" not in step
+            and "wait" not in step
+            and "group" not in step
+            and step.get("id")
+            not in (
+                "coverage-pr-analyze",
+                "analyze",
+                "build-x86_64",
```

Comment:
`not in "build-"`

Reply:
I wanted to be explicit about what builds we have for now. I'm also not expecting many to be added, but let's see.